### DISCUSSION TITLE
File permissions of outputted webps from mogrify command

### DISCUSSION THREAD

**Discussion by luboslives** [ImageMagick/ImageMagick#2866] [Help]
I'm not sure where the problem lies, but when I run a shell script which includes a `mogrify` command to create WebP copies of existing images, the outputted files only get created with a permission of `rw` for the owner. I'm trying to get them to `664` or `rw-rw-r--` but no luck.

**Platform**
CentOS 8
ImageMagick.x86_64 6.9.10.86-1 from the EPEL repo
Running [watchexec](https://github.com/watchexec/watchexec) to watch an uploads directory.
Running a bash script with `image_optim` and then `mogrify` commands.

The `mogrify` command is being run by a vhost's owning user, let's say `vhostowner`, and the user has the proper `umask` of `002` - when I touch a new file as the user the permissions are exactly what I need them to be. When a file gets uploaded, the permissions are also correct.

Here's the pertinent part of the shell script:

```
# image_optim command precedes this

su vhostowner -c "nice -n 15 mogrify -format webp -define webp:method=6 -define webp:filter-type=1 -define webp:filter-strength=0 -define webp:target-psnr=43 $filenames"

# trying to force permissions also has no effect
chmod 664 $filenames

# end of script
```

So i'm at a loss here, it's not obeying the user's `umask` or the `chmod` command.

**Comment by urban-warrior** [comment]
ImageMagick opens a file for writing with default permissions that respects the umask setting.  If you are not getting expected results, it likely is due to your environment.

**Reply by luboslives** [reply]
Thanks @urban-warrior, good to know it's nothing to do with `mogrify`.

If anyone else runs into this problem, I switched from the `su <user> -c` convention to `sudo -u <user>` followed by the command and that did the trick (preserved the correct file permissions).

**Addendum:** actually it seems to now be taking on the umask / permissions of the root user instead of the owning user, but in my case that's good enough 🤷‍♂️

**Comment by urban-warrior** [comment]
ImageMagick opens an image file with default permissions.  It respects the umask settings.  If you are getting unexpected results, its likely due to something in your environment.
